[Newton] Adds policy transfer script for sim2sim transfer from Newton to physX

Milad-Rakhsha-NV opened the pull request:
# Description

This PR adds a play script to physX-based IsaacLab to make it possible to play a Newton-based trained policy. Tested environments are H1/G1/Anymal-D but other exisiting Newton environment should transfer as well.


<!--
Thank you for your interest in sending a pull request. Please make sure to check the contribution guidelines.

Link: https://isaac-sim.github.io/IsaacLab/main/source/refs/contributing.html

💡 Please try to keep PRs small and focused. Large PRs are harder to review and merge.
-->

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.
List any dependencies that are required for this change.

Fixes # (issue)

<!-- As a practice, it is recommended to open an issue to have discussions on the proposed pull request.
This makes it easier for the community to keep track of what is being developed or added, and if a given feature
is demanded by more than one party. -->

## Type of change

<!-- As you go through the list, delete the ones that are not applicable. -->

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (existing functionality will not work without user modification)
- Documentation update

## Screenshots

Please attach before and after screenshots of the change if applicable.

<!--
Example:

| Before | After |
| ------ | ----- |
| _gif/png before_ | _gif/png after_ |

To upload images to a PR -- simply drag and drop an image while in edit mode and it should upload the image directly. You can then paste that source into the above before/after sections.
-->

## Checklist

- [ ] I have read and understood the [contribution guidelines](https://isaac-sim.github.io/IsaacLab/main/source/refs/contributing.html)
- [ ] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [ ] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

<!--
As you go through the checklist above, you can mark something as done by putting an x character in it

For example,
- [x] I have done this task
- [ ] I have not done this task
-->

kellyguo11 commented on scripts/newton_sim2sim/rsl_rl_transfer.py:
```diff
@@ -0,0 +1,239 @@
+    log_root_path = os.path.join("logs", "rsl_rl", agent_cfg.experiment_name)
+    log_root_path = os.path.abspath(log_root_path)
+    print(f"[INFO] Loading experiment from directory: {log_root_path}")
+    if args_cli.use_pretrained_checkpoint:
```
is this still applicable for sim2sim in this context? we only provide physx pretrained checkpoints right now, so maybe this only makes sense if we are running newton?

kellyguo11 commented on scripts/newton_sim2sim/rsl_rl_transfer.py:
```diff
@@ -0,0 +1,239 @@
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Script to play a checkpoint of an RL agent from RSL-RL with policy transfer capabilities."""
```
I think it'll be good to add this into the documentation as well. right now, we only provide commands for running physx trained policies in newton

Milad-Rakhsha-NV replied:
sounds good, let's decide on the structure of these scripts and yaml and whether it should be consistent with dev/newton ones and I will take care of this

kellyguo11 commented on scripts/newton_sim2sim/policy_mapping.py:
```diff
@@ -0,0 +1,74 @@
+and provides observation and action remapping functions.
+"""
+
+import yaml
```
in general, we want to keep the scripts folder to only contain python scripts that can be executed directly. For simplicity, maybe we can just move this into the rsl_rl_transfer.py script?

Milad-Rakhsha-NV replied:
yep will do that

kellyguo11 commented on scripts/newton_sim2sim/mappings/sim2sim_anymal_d.yaml:
```diff
@@ -0,0 +1,34 @@
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+# Joint names in the source physics engine where policy is trained (Newton)
```
is this the temporary method for specifying these joint names, eventually I think we'd want to parse this from the robot schema or enforce some ordering between physx and newton?

also, since they specify source and target, maybe in the file name, instead of sim2sim, we specify explicitly newton2physx

Milad-Rakhsha-NV replied:
robot schema is goign to be a bit more complex but is doable and we can look into later

kellyguo11 commented on scripts/newton_sim2sim/mappings/sim2sim_anymal_d.yaml:
```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2022-2025, The Isaac Lab Project Developers (https://github.com/isaac-sim/IsaacLab/blob/main/CONTRIBUTORS.md).
```
might just be a personal taste, but I'm not a fan of "mappings" as the folder name. I would propose to just go with something simple like config

kellyguo11 commented:
ok let's fix what we can for now. please also run the formatter to resolve the pre-commit failures

Milad-Rakhsha-NV commented:
@kellyguo11 I updated the documentation as well. 
I updated the feature/dev branch for consistency as well please see this 
https://github.com/isaac-sim/IsaacLab/pull/3581
